app/grandchallenge/algorithms/models.py

```diff
@@ -702,6 +710,10 @@ def save(self, *args, **kwargs):
                 actor=self.user,
                 target=self.base_object,
             )
+        elif adding and not self.algorithm.require_user_review:
+            # immediately allow access, no need for a notification
+            self.status = self.ACCEPTED
+            self.save()
```

Comment:
Setting it up this way you call save recursively, it is better to avoid that and move conditionally setting `self.status` to before the first call to `super().save(*args, **kwargs)` on L702.

Reply:
This happens in a few other places so change those too.

Reply:
Ah right, will fix that and I will also add tests. Thanks!